src/core/activityContent/validation.ts

```diff
@@ -40,386 +42,388 @@ const SUPPORTED_IMAGE_MEDIA_TYPES = [
+const isValidDurationField = (obj: string): boolean => obj.match(DURATION_REGEX) !== null;
 
-  if (obj["height"] && !isNumber(obj["height"])) return false;
-  if (obj["width"] && !isNumber(obj["width"])) return false;
+const hasAtLeastOneSupportedAudioMediaType = (obj: Array<ActivityContentAudioLink>): boolean => {
+  for (const audioLink of obj) {
+    if (SUPPORTED_AUDIO_MEDIA_TYPES.includes(audioLink["mediaType"])) return true;
+  }
 
-  return true;
+  return false;
 };
 
-const isActivityContentHash = (obj: unknown): obj is ActivityContentHash => {
-  if (!isRecord(obj)) return false;
-  if (!isString(obj["algorithm"])) return false;
-  if (!isString(obj["value"])) return false;
+const hasAtLeastOneSupportedImageMediaType = (obj: Array<ActivityContentImageLink>): boolean => {
+  for (const imageLink of obj) {
+    if (SUPPORTED_IMAGE_MEDIA_TYPES.includes(imageLink["mediaType"])) return true;
+  }
 
-  return true;
+  return false;
 };
 
-const isActivityContentLocation = (obj: unknown): obj is ActivityContentLocation => {
-  if (!isRecord(obj)) return false;
-  if (obj["type"] !== "Place") return false;
-
-  if (obj["name"] && !isString(obj["name"])) return false;
-  if (obj["accuracy"] && !isNumber(obj["accuracy"])) return false;
-  if (obj["altitude"] && !isNumber(obj["altitude"])) return false;
-  if (obj["latitude"] && !isNumber(obj["latitude"])) return false;
-  if (obj["longitude"] && !isNumber(obj["longitude"])) return false;
-  if (obj["radius"] && !isNumber(obj["radius"])) return false;
-  if (obj["units"] && !isString(obj["units"])) return false;
+const hasAtLeastOneSupportedVideoMediaType = (obj: Array<ActivityContentVideoLink>): boolean => {
+  for (const videoLink of obj) {
+    if (SUPPORTED_VIDEO_MEDIA_TYPES.includes(videoLink["mediaType"])) return true;
+  }
 
-  return true;
+  return false;
 };
 
-const isActivityContentTag = (obj: unknown): obj is ActivityContentTag =>
-  isActivityContentHashtag(obj) || isActivityContentMention(obj);
-
-const isActivityContentHashtag = (obj: unknown): obj is ActivityContentHashtag => {
-  if (!isRecord(obj)) return false;
-  if (obj["type"] !== undefined) return false;
-  if (!isString(obj["name"])) return false;
+const hasAtLeastOneSupportedHashAlgorithm = (obj: Array<ActivityContentHash>): boolean => {
+  for (const hash of obj) {
+    if (hash["algorithm"] === "keccak256" && isString(hash["value"]) && hash["value"].match(HASH_REGEX)) return true;
+  }
 
-  return true;
+  return false;
 };
 
-const isActivityContentMention = (obj: unknown): obj is ActivityContentMention => {
-  if (!isRecord(obj)) return false;
-  if (obj["type"] !== "Mention") return false;
-  if (!isDSNPUserId(obj["id"])) return false;
-
-  if (obj["name"] && !isString(obj["name"])) return false;
+const getActivityContentAttachment = (obj: unknown): ActivityContentAttachment | undefined => {
+  try {
+    return requireActivityContentAttachment(obj);
+  } catch (e) {
```

Comment:
Can we change this to specifically catch `InvalidActivityContentError` so it doesn't catch real bugs in case we screw something up in the future?